carbon-design-system/carbon-for-ibm-dotcom
# fix(Masthead): set selected state for L1 menu items and mobile

### Related Ticket(s)

[Masthead L1]: Url selected state logic not applied to Masthead L1 #7312

### Description

The selected state for menu items based off URL hasn't been implemented for React Masthead L1 and mobile. This PR iterates through the menu items to set the state for the parent menu item and sets the selected state for every menu item in the hierarchy for mobile.

DESKTOP:
<img width="1368" alt="Screen Shot 2021-10-27 at 6 57 06 PM" src="https://user-images.githubusercontent.com/54281166/139159474-c993f1d8-63de-46ef-a2a1-458f02ee3a0f.png">

MOBILE:
![Oct-27-2021 18-58-54](https://user-images.githubusercontent.com/54281166/139159513-8a1e0bc2-24e3-478f-87e4-1e33027dec5d.gif)

**TO TEST:**
1. Go to this codesandbox: https://codesandbox.io/s/sad-mclaren-1vw1z
2. Click on any menu item and see the selected state appear

### Changelog

**New**

- add functions for iterating through menu items to check for url
- add selected prop/set classname for mobile menu items

**Changed**

- be more specific in querying `lastMenuItem` and `platform` for the `onBlur` for the hamburger menu
